backend/src/main/java/com/javahelp/backend/TokenAuthResult.java

```diff
@@ -0,0 +1,36 @@
+package com.javahelp.backend;
+
+import com.javahelp.model.token.Token;
+import com.javahelp.model.user.User;
+
+import org.graalvm.compiler.api.replacements.Snippet;
+
+public class TokenAuthResult {
+    private boolean authenticated;
+    private User user;
+    private Token token;
```

Comment:
I am thinking if it will be better to set user and token (and even authenticated?) to final as well, since they are associated with only one user and the corresponding token? (Or even errorMessage to be final as well? Since if this authentication fails, the user has to initiate another login request and it will be another `TokenAuthResult` generated.) Just some considerations.